### ISSUE TITLE
Rule Change: Add more `allow*` options to `use-baseline` rule

### ISSUE THREAD

**Issue by thecalamiity** [eslint/css#298]
### What rule do you want to change?

use-baseline

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Add options to allow specific CSS functions, media conditions, and property values to be exempt from baseline enforcement.

### Example code

```css
.selector {
  color: color-mix(in srgb, plum, #123456);
  clip-path: fill-box;
}

@media (scripting: none) {}
```

### What does the rule currently do for this code?

It reports that:
- The `color-mix()` function is not a widely available baseline feature.
- The value `fill-box` for `clip-path` is not a widely available baseline feature.
- The media condition (`scripting: none`) is not a widely available baseline feature.

Currently, these cannot be selectively allowed.

### What will the rule do after it's changed?

If configured like this:

```js
"css/use-baseline": [
  "error",
  {
    "available": "widely",
    "allowFunctions": ["color-mix"],
    "allowMediaConditions": ["scripting"],
    "allowPropertyValues": {
      "clip-path": ["fill-box"]
    }
  }
]
```

then the rule would not report errors for the above code.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Sounds good to me. :+1: Marking as accepted.
